### ISSUE TITLE
[bug] windows-2022 can no longer run applications compiled with included mingw64

### ISSUE THREAD

**Issue by IsaacShelton** [actions/runner-images#6412]
### Description

When you run a C/C++ executable compiled with the included mingw64, you would expect it to run without any issues on the same machine, instead of ~~silently~~ failing.

My particular workflow was working completely fine, but now, without any changes to my workflow, everything I build with mingw64 on Github runners ~~silently~~ crashes immediately.

### Solution:

It's probably an issue with the MinGW DLLs not being included in the path.

(not having access or having outdated versions of MinGW DLLs can cause the executable to crash) 

MinGW DLLs live at `C:\ProgramData\chocolatey\lib\mingw\tools\install\mingw64\bin`.

### Benefits of keeping change
Not having the DLLs in the path would make for more realistic testing.

### Consequences of keeping change
The drawback is that you can't use mingw64 to easily test C/C++ code without manually changing the path yourself or searching for and copying the DLLs before running your code. This requirement is not made clear in build logs ~~(completely silent error)~~.

### Platforms affected

- [ ] Azure DevOps
- [X] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 10.15
- [ ] macOS 11
- [ ] macOS 12
- [ ] Windows Server 2019
- [X] Windows Server 2022

### Image version and build link

Broken as of 2.298.2

### Is it regression?

Yes, but hard to tell what runner version it broke due to caching.

My cache recently expired, so it could be any version after `windows-latest` Jun 8.

`windows-latest` Jun 8 - last successful build with cache miss

`windows-2022` 2.297.0 - last successful build with cache hit

### Expected behavior

When you run a C/C++ executable compiled with the included mingw64, you would expect it to run without any issues on the same machine, instead of ~~silently~~ failing.

### Actual behavior

In a workflow, attempting run a C/C++ executable compiled with the included mingw64 ~~silently~~ fails (even if done immediately after building).

### Repro steps

1. [workflow job] Compile basic C/C++ program using included mingw64
2. [workflow job] Run resulting executable
3. (it will fail silently - the workflow will continue as if nothing went wrong - the executable will crash immediately)

# Edit:

This doesn't solve the issue of the resulting executables immediately crashing, but a slight correction:

Maybe the workflow will stop. The executable would crash with an exitcode of something like -12345.  It's just that it doesn't stop my workflow because of how I invoke them.

TL;DR - The runner cannot run executables compiled with the included mingw64. If you try to run a resulting executable, the executable will instantly crash.

**Comment by al-cheb** [comment]
Hey @IsaacShelton.
We will take a look at it.

**Comment by al-cheb** [comment]
Close as https://github.com/actions/runner-images/issues/6103
